ottobackwards/metron
# Refactor hdfs and pattern loading to work with extensions

Create GrokBuilder and ResourceLoader for re-usable and consistant file loading and path fixups
tests create the correct ./target environment for rules to be loaded from

ParserBolt injects the global configuration into the parser configuration
Global configuration has new key for metron hdfs root

new key is setup by ambari

Now, give a parser configuration with a simple grokPath "/patterns/yaf"
- it will be fixed to work in hdfs or local for tests
- it will support yaf as a directory not just a file

##Testing
spin up full dev
open metron configuration ui

You should be able to
 - open yaf and edit/save, reopen see changes to grok patterns
 - create a new grok parser and do the same


## Contributor Comments
[Please place any comments here.  A description of the problem/enhancement, how to reproduce the issue, your testing methodology, etc.]


## Pull Request Checklist

Thank you for submitting a contribution to Apache Metron.  
Please refer to our [Development Guidelines](https://cwiki.apache.org/confluence/pages/viewpage.action?pageId=61332235) for the complete guide to follow for contributions.  
Please refer also to our [Build Verification Guidelines](https://cwiki.apache.org/confluence/display/METRON/Verifying+Builds?show-miniview) for complete smoke testing guides.  


In order to streamline the review of the contribution we ask you follow these guidelines and ask you to double check the following:

### For all changes:


### For code changes:
  ```
  mvn -q clean integration-test install && build_utils/verify_licenses.sh 
  ```


### For documentation related changes:

  ```
  cd site-book
  mvn site
  ```

#### Note:
Please ensure that once the PR is submitted, you check travis-ci for build issues and submit an update to your PR as soon as possible.
It is also recommended that [travis-ci](https://travis-ci.org) is set up for your personal repository such that your branches are built there before submitting a pull request.


---
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/ottobackwards/metron/6)
